Tasks/GradleV3/gradletask.ts

```diff
@@ -221,96 +51,111 @@ async function run() {
 
         // Enable SonarQube Analysis (if desired)
-        let isSonarQubeEnabled: boolean = tl.getBoolInput('sqAnalysisEnabled', false);
+        const isSonarQubeEnabled: boolean = tl.getBoolInput('sqAnalysisEnabled', false);
         if (isSonarQubeEnabled) {
             // Looks like: 'SonarQube analysis is enabled.'
             console.log(tl.loc('codeAnalysis_ToolIsEnabled'), 'SonarQube');
-            gradleRunner = <ToolRunner> sqGradle.applyEnabledSonarQubeArguments(gradleRunner);
+            gradleRunner = <ToolRunner>sqGradle.applyEnabledSonarQubeArguments(gradleRunner);
         }
         gradleRunner = codeAnalysisOrchestrator.configureBuild(gradleRunner);
 
         // START: Run code analysis
-        let gradleResult: number;
-        let statusFailed: boolean = false;
-        let analysisError: any;
+        const codeAnalysisResult: ICodeAnalysisResult = {};
+
         try {
-            gradleResult = await gradleRunner.exec(getExecOptions());
-            tl.debug(`Gradle result: ${gradleResult}`);
+            codeAnalysisResult.gradleResult = await gradleRunner.exec(getExecOptions());
+            codeAnalysisResult.statusFailed = false;
+            codeAnalysisResult.analysisError = '';
+
+            tl.debug(`Gradle result: ${codeAnalysisResult.gradleResult}`);
         } catch (err) {
+            codeAnalysisResult.gradleResult = -1;
+            codeAnalysisResult.statusFailed = true;
+            codeAnalysisResult.analysisError = err;
+
             console.error(err);
             tl.debug('taskRunner fail');
-            gradleResult = -1;
-            statusFailed = true;
-            analysisError = err;
         }
+
         tl.debug('Processing code analysis results');
         codeAnalysisOrchestrator.publishCodeAnalysisResults();
 
         // We should always publish test results and code coverage
         publishTestResults(publishJUnitResults, testResultsFiles);
-        await publishCodeCoverage(isCodeCoverageOpted, failIfCodeCoverageEmpty, codeCoverageTool, summaryFile, reportDirectory);
 
-        if (gradleResult === 0) {
-            tl.setResult(tl.TaskResult.Succeeded, 'Build succeeded.');
-        } else if (gradleResult === -1 && statusFailed === true) {
-            tl.setResult(tl.TaskResult.Failed, analysisError);
-        } else {
-            tl.setResult(tl.TaskResult.Failed, 'Build failed.');
-        }
+        const publishCodeCoverageSettings: IPublishCodeCoverageSettings = {
+            isCodeCoverageOpted: isCodeCoverageOpted,
+            failIfCoverageEmpty: failIfCodeCoverageEmpty,
+            codeCoverageTool: codeCoverageTool,
+            summaryFile: summaryFile,
+            reportDirectory: reportDirectory
+        };
+        await publishCodeCoverageResultsAsync(publishCodeCoverageSettings);
+
+        const taskResult: ITaskResult = resolveTaskResult(codeAnalysisResult);
```

Comment:
Could it make sense to add input to be able to turn off this logic? Is this backward compatible?

Reply:
I double-checked my changes and I believe that we don't have to hide this functionality behind the task input, the changes themselves are backward compatible.